core/src/test/java/io/grpc/DnsNameResolverTest.java

```diff
@@ -149,14 +149,14 @@ public void resolve() throws Exception {
     assertEquals(1, fakeExecutor.runDueTasks());
     verify(mockListener).onUpdate(resultCaptor.capture(), any(Attributes.class));
     assertEquals(name, resolver.invocations.poll());
-    assertAnswerMatches(answer1, 81, resultCaptor.getValue());
+    assertAnswerMatches(answer1, 81, resultCaptor.getValue().get(0));
```

Comment:
Use `Iterables.getOnlyElement(resultCaptor.getValue())` so you can also verify that the top-level list only has one element.